flycheck-vale.el

```diff
@@ -51,97 +51,42 @@
-rest (e.g. filename) gets filled in elsewhere."
-  (let-alist issue
-    (flycheck-error-new
-     :line .Line
-     :column (elt .Span 0)
-     :message .Message
-     :level (assoc-default .Severity flycheck-vale--level-map 'string-equal 'error))))
-
-(defun flycheck-vale--output-to-errors (output)
+(defun flycheck-vale--output-to-errors (output checker buffer)
   "Parse the full JSON output of vale, OUTPUT, into a sequence of flycheck error structs."
-  (let* ((full-results (json-read-from-string output))
-
-         ;; Get the list of issues for each file.
-         (result-vecs (mapcar 'cdr full-results))
-
-         ;; Chain all of the issues together. The point here, really, is that we
-         ;; don't expect results from more than one file, but we should be
-         ;; prepared for the theoretical possibility that the issues are somehow
-         ;; split across multiple files. This is basically a punt in lieu of
-         ;; more information.
-         (issues (apply 'append (mapcar 'cdr full-results))))
-    (mapcar 'flycheck-vale--issue-to-error issues)))
-
-(defun flycheck-vale--handle-finished (checker callback buf)
-  "Parse the contents of the output buffer into flycheck error
-structures, attaching CHECKER and BUF to the structures, and
-passing the results to CALLBACK."
-  (let* ((output (with-current-buffer flycheck-vale-output-buffer (buffer-string)))
-         (errors (flycheck-vale--output-to-errors output)))
-    ;; Fill in the rest of the error struct database
-    (cl-loop for err in errors do
-          (setf
-           (flycheck-error-buffer err) buf
-           (flycheck-error-filename err) (buffer-file-name buf)
-           (flycheck-error-checker err) checker))
-    (funcall callback 'finished errors)))
-
-(defun flycheck-vale--normal-completion? (event)
-  (or  (string-equal event "finished\n")
-       (string-match "exited abnormally with code 1.*" event)))
-
-(defun flycheck-vale--start (checker callback)
-  "Run vale on the current buffer's contents with CHECKER, passing the results to CALLBACK."
-
-  ;; Clear the output buffer
-  (with-current-buffer (get-buffer-create flycheck-vale-output-buffer)
-    (read-only-mode 0)
-    (erase-buffer))
-
-  (let* ((process-connection-type nil)
-         (proc (start-process "flycheck-vale-process"
-                              flycheck-vale-output-buffer
-                              flycheck-vale-program
-                              "--output"
-                              "JSON")))
-    (lexical-let ((checker checker)
-                  (callback callback)
-                  (buf (current-buffer)))
-      (set-process-sentinel
-       proc
-       #'(lambda (process event)
-           (when (flycheck-vale--normal-completion? event)
-             (flycheck-vale--handle-finished checker callback buf)))))
-
-    (process-send-region proc (point-min) (point-max))
-    (process-send-eof proc)))
+  (let* ((issues (cdar (json-read-from-string output)))
```

Comment:
I'm a bit concerned about this. It assume that all of the issues will be reported in the first entry of the `output` list. While this is probably true, it feels intellectually lazy to ignore the rest of the entries, should vale choose to (for whatever reason) split them across multiple entries.

My original code read all of the entries, with a substantial comment explaining why. I can't find any vale documentation describing the JSON format, so I'd like to be as comprehensive as possible.